Memory growth is now the same across all 8 available GPUs.
Loading data...
Video 0 ...
Video 50 ...
Video 100 ...
Video 150 ...
Video 200 ...
Video 250 ...
Video 300 ...
Video 350 ...
Done loading videos in 62.839847803115845 seconds.
Splitting videos into train, val, and test...
2022-08-06 14:44:28.145415: I tensorflow/core/platform/cpu_feature_guard.cc:152] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE3 SSE4.1 SSE4.2 AVX
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-08-06 14:44:34.790499: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 14649 MB memory:  -> device: 0, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:06:00.0, compute capability: 7.0
2022-08-06 14:44:34.794232: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 14649 MB memory:  -> device: 1, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:07:00.0, compute capability: 7.0
2022-08-06 14:44:34.796753: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:2 with 14649 MB memory:  -> device: 2, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:0a:00.0, compute capability: 7.0
2022-08-06 14:44:34.799141: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:3 with 14649 MB memory:  -> device: 3, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:0b:00.0, compute capability: 7.0
2022-08-06 14:44:34.802817: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:4 with 14649 MB memory:  -> device: 4, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:85:00.0, compute capability: 7.0
2022-08-06 14:44:34.805200: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:5 with 14649 MB memory:  -> device: 5, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:86:00.0, compute capability: 7.0
2022-08-06 14:44:34.807564: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:6 with 14649 MB memory:  -> device: 6, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:89:00.0, compute capability: 7.0
2022-08-06 14:44:34.809889: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:7 with 14649 MB memory:  -> device: 7, name: Tesla V100-SXM2-16GB-N, pci bus id: 0000:8a:00.0, compute capability: 7.0
Done splitting.
Setting TensorFlow Mirrored Strategy with 1 GPUs...
Number of devices in strategy: 1
Creating TF Dataset...
Creating TF DISTRIBUTED Dataset...
Creating CNN Model on Each Active Replica (GPU)
2022-08-06 14:44:51.248842: W tensorflow/core/grappler/optimizers/data/auto_shard.cc:776] AUTO sharding policy will apply DATA sharding policy as it failed to apply FILE sharding policy because of the following reason: Found an unshardable source dataset: name: "TensorSliceDataset/_2"
op: "TensorSliceDataset"
input: "Placeholder/_0"
input: "Placeholder/_1"
attr {
  key: "Toutput_types"
  value {
    list {
      type: DT_UINT8
      type: DT_UINT8
    }
  }
}
attr {
  key: "_cardinality"
  value {
    i: 232
  }
}
attr {
  key: "is_files"
  value {
    b: false
  }
}
attr {
  key: "metadata"
  value {
    s: "\n\024TensorSliceDataset:0"
  }
}
attr {
  key: "output_shapes"
  value {
    list {
      shape {
        dim {
          size: 461
        }
        dim {
          size: 224
        }
        dim {
          size: 224
        }
        dim {
          size: 3
        }
      }
      shape {
      }
    }
  }
}
experimental_type {
  type_id: TFT_PRODUCT
  args {
    type_id: TFT_DATASET
    args {
      type_id: TFT_PRODUCT
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
    }
  }
  args {
    type_id: TFT_DATASET
    args {
      type_id: TFT_PRODUCT
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
    }
  }
}
Downloading data from https://storage.googleapis.com/tensorflow/keras-applications/vgg19/vgg19_weights_tf_dim_ordering_tf_kernels_notop.h5
80142336/80134624 [==============================] - 2s 0us/step
80150528/80134624 [==============================] - 2s 0us/step
Beginning Feature Extraction in GPU Mode...
2022-08-06 14:45:01.435741: W tensorflow/core/kernels/gpu_utils.cc:50] Failed to allocate memory for convolution redzone checking; skipping this check. This is benign and only means that we won't check cudnn for out-of-bounds reads and writes. This message will only be printed once.
2022-08-06 14:45:02.521331: I tensorflow/stream_executor/cuda/cuda_dnn.cc:379] Loaded cuDNN version 8400
2022-08-06 14:45:03.787972: W tensorflow/core/common_runtime/bfc_allocator.cc:343] Garbage collection: deallocate free memory regions (i.e., allocations) so that we can re-allocate a larger region to avoid OOM due to memory fragmentation. If you see this message frequently, you are running near the threshold of the available device memory and re-allocation may incur great performance overhead. You may try smaller batch sizes to observe the performance impact. Set TF_ENABLE_GPU_GARBAGE_COLLECTION=false if you'd like to disable this feature.
Done getting video frame feature representations in 272.5213027000427 seconds.
Formatting Results into Numpy Arrays...
Setting TensorFlow Mirrored Strategy with 1 GPUs...
Number of devices in strategy: 1
Creating TF Dataset...
Creating TF DISTRIBUTED Dataset...
Creating CNN Model on Each Active Replica (GPU)
2022-08-06 14:49:28.860765: W tensorflow/core/grappler/optimizers/data/auto_shard.cc:776] AUTO sharding policy will apply DATA sharding policy as it failed to apply FILE sharding policy because of the following reason: Found an unshardable source dataset: name: "TensorSliceDataset/_2"
op: "TensorSliceDataset"
input: "Placeholder/_0"
input: "Placeholder/_1"
attr {
  key: "Toutput_types"
  value {
    list {
      type: DT_UINT8
      type: DT_UINT8
    }
  }
}
attr {
  key: "_cardinality"
  value {
    i: 59
  }
}
attr {
  key: "is_files"
  value {
    b: false
  }
}
attr {
  key: "metadata"
  value {
    s: "\n\024TensorSliceDataset:1"
  }
}
attr {
  key: "output_shapes"
  value {
    list {
      shape {
        dim {
          size: 461
        }
        dim {
          size: 224
        }
        dim {
          size: 224
        }
        dim {
          size: 3
        }
      }
      shape {
      }
    }
  }
}
experimental_type {
  type_id: TFT_PRODUCT
  args {
    type_id: TFT_DATASET
    args {
      type_id: TFT_PRODUCT
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
    }
  }
  args {
    type_id: TFT_DATASET
    args {
      type_id: TFT_PRODUCT
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
    }
  }
}
Beginning Feature Extraction in GPU Mode...
Done getting video frame feature representations in 62.98982787132263 seconds.
Formatting Results into Numpy Arrays...
Setting TensorFlow Mirrored Strategy with 1 GPUs...
Number of devices in strategy: 1
Creating TF Dataset...
Creating TF DISTRIBUTED Dataset...
Creating CNN Model on Each Active Replica (GPU)
Beginning Feature Extraction in GPU Mode...
2022-08-06 14:50:34.667516: W tensorflow/core/grappler/optimizers/data/auto_shard.cc:776] AUTO sharding policy will apply DATA sharding policy as it failed to apply FILE sharding policy because of the following reason: Found an unshardable source dataset: name: "TensorSliceDataset/_2"
op: "TensorSliceDataset"
input: "Placeholder/_0"
input: "Placeholder/_1"
attr {
  key: "Toutput_types"
  value {
    list {
      type: DT_UINT8
      type: DT_UINT8
    }
  }
}
attr {
  key: "_cardinality"
  value {
    i: 73
  }
}
attr {
  key: "is_files"
  value {
    b: false
  }
}
attr {
  key: "metadata"
  value {
    s: "\n\024TensorSliceDataset:2"
  }
}
attr {
  key: "output_shapes"
  value {
    list {
      shape {
        dim {
          size: 461
        }
        dim {
          size: 224
        }
        dim {
          size: 224
        }
        dim {
          size: 3
        }
      }
      shape {
      }
    }
  }
}
experimental_type {
  type_id: TFT_PRODUCT
  args {
    type_id: TFT_DATASET
    args {
      type_id: TFT_PRODUCT
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
    }
  }
  args {
    type_id: TFT_DATASET
    args {
      type_id: TFT_PRODUCT
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
      args {
        type_id: TFT_TENSOR
        args {
          type_id: TFT_UINT8
        }
      }
    }
  }
}
Done getting video frame feature representations in 77.85544443130493 seconds.
Formatting Results into Numpy Arrays...
Back from feature Extraction.
Train Features: (232, 461, 512) || Train Labels: (232, 1)
Val Features: (59, 461, 512) || Val Labels: (59, 1)
Test Features: (73, 461, 512) || Test Labels: (73, 1)
Splitting + batching features and labels for RNN ...
Training RNN ...
Epoch 1/15

8/8 [==============================] - 12s 518ms/step - loss: 0.7668 - accuracy: 0.6207 - val_loss: 0.6354 - val_accuracy: 0.6780
Epoch 2/15
8/8 [==============================] - 2s 196ms/step - loss: 0.4678 - accuracy: 0.7845 - val_loss: 0.4876 - val_accuracy: 0.7627
Epoch 3/15
8/8 [==============================] - 1s 194ms/step - loss: 0.3082 - accuracy: 0.8922 - val_loss: 0.4465 - val_accuracy: 0.8305
Epoch 4/15
8/8 [==============================] - 1s 192ms/step - loss: 0.2209 - accuracy: 0.9224 - val_loss: 0.4438 - val_accuracy: 0.8305
Epoch 5/15
8/8 [==============================] - 1s 192ms/step - loss: 0.1234 - accuracy: 0.9655 - val_loss: 0.5920 - val_accuracy: 0.7458
Epoch 6/15
8/8 [==============================] - 2s 198ms/step - loss: 0.0740 - accuracy: 0.9828 - val_loss: 0.5768 - val_accuracy: 0.8475
Epoch 7/15
8/8 [==============================] - 1s 191ms/step - loss: 0.0520 - accuracy: 0.9784 - val_loss: 0.6437 - val_accuracy: 0.7966
Epoch 8/15
8/8 [==============================] - 2s 194ms/step - loss: 0.0661 - accuracy: 0.9698 - val_loss: 0.8326 - val_accuracy: 0.7288
Epoch 9/15
8/8 [==============================] - 1s 192ms/step - loss: 0.0825 - accuracy: 0.9655 - val_loss: 0.8134 - val_accuracy: 0.7288
Epoch 10/15
8/8 [==============================] - 2s 198ms/step - loss: 0.0461 - accuracy: 0.9871 - val_loss: 0.5921 - val_accuracy: 0.8814
Epoch 11/15
8/8 [==============================] - 2s 199ms/step - loss: 0.0515 - accuracy: 0.9741 - val_loss: 0.8390 - val_accuracy: 0.6949
Epoch 12/15
8/8 [==============================] - 2s 195ms/step - loss: 0.0313 - accuracy: 0.9957 - val_loss: 0.7957 - val_accuracy: 0.8136
Epoch 13/15
8/8 [==============================] - 2s 202ms/step - loss: 0.0195 - accuracy: 0.9914 - val_loss: 1.1364 - val_accuracy: 0.7797
Epoch 14/15
8/8 [==============================] - 2s 195ms/step - loss: 0.0102 - accuracy: 1.0000 - val_loss: 0.8531 - val_accuracy: 0.8305
Epoch 15/15
8/8 [==============================] - 2s 199ms/step - loss: 0.0086 - accuracy: 0.9957 - val_loss: 0.8858 - val_accuracy: 0.8305
3/3 [==============================] - 0s 53ms/step - loss: 1.0602 - accuracy: 0.7397
Traceback (most recent call last):
  File "train_v2.py", line 410, in <module>
    main()
  File "train_v2.py", line 404, in main
    print_final_info(args.cnn_model, loss, accuracy, f1, train_feature_duration)
TypeError: print_final_info() missing 1 required positional argument: 'val_duration'